### PR TITLE
Python, package, and auth upgrades

### PR DESCRIPTION
**Pull request by zzzeid**
PR for following bugs:
https://bugzilla.mozilla.org/show_bug.cgi?id=1615411
https://bugzilla.mozilla.org/show_bug.cgi?id=1614433
https://bugzilla.mozilla.org/show_bug.cgi?id=1614431

General upgrades:
- use python3.8 in Dockerfiles
- upgrade flask, cffi, cryptography, flask-wtf
- add requirements.in file for pip-compile to use

Authentication:
- removed package downgrade requirement for pyoidc
- updated implementation to work with latest flask-pyoidc and pyoidc
- implemented custom classes and methods to work around Auth0 bug

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -24,16 +24,6 @@ STUFF:
...
 jobs:
-  lint:
-    working_directory: /app
-    docker:
-      - *python_image
-    steps:
-      - checkout
-      - run: pip install yapf==0.16.1 && pip install flake8==3.5.0 && pip install pytest==3.6.3
-      - run: yapf --diff --recursive ./
-      - run: flake8 .
```
Comment (globau) [inline .circleci/config.yml]: looks like this overlaps with #99, which i call out that linting tests shouldn't be removed from our CI runs.

File: tests/test_py3_codequality.py [hunk 2]
```diff
@@ -5,17 +5,3 @@
...
-    cmd = ('yapf', '--diff', '--recursive', '--exclude', './.git/*')
-    passed = len(subprocess.check_output(cmd + files)) == 0
-    assert passed, 'The python code does not adhear to the project style.'
-
-
-def test_check_python_flake8():
-    files = ('.', )
-    cmd = ('flake8', )
-    passed = subprocess.call(cmd + files) == 0
-    assert passed, 'Flake8 did not run cleanly.'
```
Comment (globau) [inline tests/test_py3_codequality.py]: more overlap with #99 - make sure this change to remove style tests doesn't land.

File: .circleci/config.yml (hunk 1 above)
Comment (zzzeid) [reply .circleci/config.yml]: @globau thanks for the feedback -- my goal is to merge #99 before this one and then rebase this branch with the updated master. In terms of removing linting from CI - please see comment on #99.

File: tests/test_py3_codequality.py (hunk 2 above)
Comment (zzzeid) [reply tests/test_py3_codequality.py]: This change will not land in master, but will be updated with what's in #99. The reason this was removed was that the updated versions of `yapf` and `flake8` conflict with each other in numerous ways and we've decided to switch to black to match with lando-api.

**Pull request merged**